### ISSUE TITLE
Separate check for possibly unused arguments if locals() is used

### ISSUE THREAD

**Issue by Stealthii** [PyCQA/pylint#4912]
### Current problem

* I have a long list of keywords to a method, that I want to define for type hints
* I need to pass these to another method as a dictionary, using `locals()`
* pylint detects possibly unused variables, but not arguments

### Desired solution

* pylint should extend the logic used around unused argument and variable definitions to consider arguments as potentially unused as well

### Additional context

Here is an example function that makes use of local scope variables and arguments.
```python
def example(arg_a='a', arg_b='b'):  # unused-argument
    """An example function."""
    var_c = 'c'  # potentially-unused-variable
    return locals()  # {'arg_a': 'a', 'arg_b': 'b', 'var_c': 'c'}
```

**Comment by Stealthii** [comment]
This is related to #1909, which implemented a check for potentially unused variables. Arguments to a function/method do not fall under this check.

I believe for consistency that a 'possibly-unused-argument' check should exist similar to 'potentially-unused-variable', regardless of opinion about `locals()` use.

**Comment by bersbersbers** [comment]
Here's an example summarizing the missing consistency with `(possibly-)unused-variable`:

```python
"""Demonstrate use case of *possibly* unused argument."""

def _fun1():
    var = 1  # pylint:disable=unused-variable

def _fun2():
    var = 1  # pylint:disable=possibly-unused-variable # note "possibly"!
    return locals()

def _fun3(arg=1):  # pylint:disable=unused-argument
    pass

def _fun4(arg=1):  # this should be a *possibly*-unused-argument
    return locals()
```
